File: mobile/lib/core/utils/image_url.dart
import '../config/api_config.dart';

String resolveImageUrl(String? url) {
  if (url == null || url.trim().isEmpty) return '';
  final trimmed = url.trim();
  if (trimmed.startsWith('http://') || trimmed.startsWith('https://')) {
    return trimmed;
  }
  if (trimmed.startsWith('/')) {
    return '${ApiConfig.origin}$trimmed';
  }
  return '${ApiConfig.origin}/$trimmed';
}

File: web/src/lib/imageUrl.ts
import { getApiOrigin } from "./utils";

export function resolveImageUrl(url?: string | null): string {
  if (!url || url.trim() === "") return "";
  const trimmed = url.trim();
  if (trimmed.startsWith("http://") || trimmed.startsWith("https://")) {
    return trimmed;
  }
  if (trimmed.startsWith("/")) {
    return `${getApiOrigin()}${trimmed}`;
  }
  return `${getApiOrigin()}/${trimmed}`;
}
